python/lsst/obs/base/instrument.py

```diff
@@ -339,6 +352,24 @@ class attribute for curated calibrations corresponding to the
             for calib, dataId in datasetRecords:
                 butler.put(calib, datasetType, dataId)
 
+    @abstractmethod
+    def makeDataIdTranslatorFactory(self) -> TranslatorFactory:
+        """Return a factory for creating Gen2->Gen3 data ID translators,
+        specialized for this instrument.
+
+        Derived class implementations should generally call
+        `TranslatorFactory.addGenericInstrumentRules` with appropriate
+        arguments, but are not required to (and may not be able to if their
+        Gen2 raw data IDs are sufficiently different from the HSC/DECam/CFHT
+        norm).
+
+        Returns
+        -------
+        factory : `TranslatorFactory`.
+            Factory for `Translator` objects.
+        """
+        raise NotImplementedError("Must be implemented by derived classes.")
```

Comment:
This is redundant with `@abstractmethod`, no? Why not just `pass`?

Reply:
I don't know.  Why not `NotImplementedError`?  I guess I just feel that's it's weird to have an abstractmethod _not_ raise, and this is what I've always seen done.

Reply:
I'm in the "abstract methods just `pass`" camp most of the time. It doesn't usually matter, since you can't have a concrete implementation without defining the abstract method, but it does let subclasses safely call `super` without raising.